File: ocaml/rna-transcription/rna_transcription.ml
(** Rna-transcription exercise *)
open Core.Std

type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let to_rna dna_list =
  let transform = function
    | `A -> `U
    | `T -> `A
    | `C -> `G
    | `G -> `C in
  List.map dna_list ~f:transform

File: fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription

let toRna(dna : string) : string = 
    let mapNucleotide n = 
        match n with
        | 'G' -> 'C'
        | 'C' -> 'G'
        | 'T' -> 'A'
        | 'A' -> 'U'
    dna.ToCharArray()
    |> Array.map mapNucleotide
    |> (fun c -> System.String c)